common/src/attestations/signers.rs

```diff
@@ -1,85 +1,93 @@
 // Copyright 2023-, Edge & Node, GraphOps, and Semiotic Labs.
 // SPDX-License-Identifier: Apache-2.0
 
-use eventuals::{Eventual, EventualExt, EventualWriter};
+use eventuals::{Eventual, EventualExt};
 use std::collections::HashMap;
 use std::sync::Arc;
 use thegraph_core::{Address, ChainId};
-use tokio::sync::watch;
 use tokio::{
     select,
-    sync::{watch::Receiver, Mutex},
+    sync::{
+        watch::{self, Receiver},
+        Mutex,
+    },
 };
 use tracing::warn;
 
 use crate::prelude::{Allocation, AttestationSigner};
 
 /// An always up-to-date list of attestation signers, one for each of the indexer's allocations.
-pub fn attestation_signers(
+pub async fn attestation_signers(
     indexer_allocations: Eventual<HashMap<Address, Allocation>>,
     indexer_mnemonic: String,
     chain_id: ChainId,
     mut dispute_manager_rx: Receiver<Option<Address>>,
-) -> Eventual<HashMap<Address, AttestationSigner>> {
+) -> Receiver<HashMap<Address, AttestationSigner>> {
     let attestation_signers_map: &'static Mutex<HashMap<Address, AttestationSigner>> =
         Box::leak(Box::new(Mutex::new(HashMap::new())));
+    let (allocations_tx, mut allocations_rx) =
+        watch::channel(indexer_allocations.value_immediate().unwrap_or_default());
+    let starter_signers_map = modify_sigers(
+        Arc::new(indexer_mnemonic.clone()),
+        chain_id,
+        attestation_signers_map,
+        allocations_rx.clone(),
+        dispute_manager_rx.clone(),
+    )
+    .await;
 
     // Whenever the indexer's active or recently closed allocations change, make sure
     // we have attestation signers for all of them.
-    let (mut signers_writer, signers_reader) =
-        Eventual::<HashMap<Address, AttestationSigner>>::new();
+    let (signers_tx, signers_rx) = watch::channel(starter_signers_map);
 
     tokio::spawn(async move {
-        // Listening to the allocation eventual and converting them to reciever.
-        // Using pipe for updation.
-        // For temporary pupose only.
-        let (allocations_tx, mut allocations_rx) =
-            watch::channel(indexer_allocations.value().await.unwrap());
+        // Actively listening to indexer_allocations to update allocations channel
+        // Temporary fix until the indexer_allocations is migrated to tokio watch
         let _p1 = indexer_allocations.pipe(move |allocatons| {
-            let _ = allocations_tx.send(allocatons);
+            allocations_tx.send(allocatons).unwrap();
         });
```

Comment:
I think this could be moved to outside tokio::spawn. Probably right after line 29, so they are close to each other.